torch/distributed/distributed_c10d.py

```diff
@@ -2139,17 +2139,28 @@ def all_gather_coalesced(
     ]
 
     if group is None:
-        default_pg = _get_default_group()
-        work = default_pg.allgather_coalesced(output_tensor_lists, input_tensor_list)
-    else:
-        work = group.allgather_coalesced(output_tensor_lists, input_tensor_list)
-
+        group = _get_default_group()
+    
+    flatten_tensors = [x.flatten().tolist() for x in input_tensor_list]
+    one_list = [x for i in flatten_tensors for x in i]
+    one_tensor = torch.Tensor(one_list)
```

Comment:
Can we use `stack()` + `flatten()` for this, to avoid creating additional Python lists?